nixos/modules/tasks/filesystems/bcachefs.nix

```diff
@@ -0,0 +1,24 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  inInitrd = any (fs: fs == "bcachefs") config.boot.initrd.supportedFilesystems;
+
+in
+
+{
+  config = mkIf (any (fs: fs == "bcachefs") config.boot.supportedFilesystems) {
+
+    system.fsPackages = [ pkgs.bcachefs-tools ];
```

Comment:
Shouldn't this also change the kernel installed until bcachefs gets upstream?

Reply:
It works now!

I will publish a tutorial on how to install NixOS with a bcachefs root partition later.

Reply:
Still has issues, don't merge yet.

Reply:
ok, I can copy the tutorial into the [wiki](nixos.wiki) later?

Reply:
sure.

the iso builds now and i can make a bcachefs partition.

good to merge :)

(i wonder why the `nixos-install` compiles the kernel again and not use the package in the iso)

Reply:
I have updated the Tutorial and linked image: https://github.com/davidak/nixos-config/blob/master/install-iso-bcachefs/README.md

Should work similar with UEFI. refer to NixOS install manual for details...